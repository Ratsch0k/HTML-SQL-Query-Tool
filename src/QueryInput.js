import React from 'react';
import * as ReactDOM from "react-dom";
import Form from "react-bootstrap/Form";
import InputGroup from "react-bootstrap/InputGroup";
import Button from "@material-ui/core/Button";
import InputAdornment from "@material-ui/core/InputAdornment";
import FormControl from "@material-ui/core/FormControl";
import Input from "@material-ui/core/Input";
import {TextField} from "@material-ui/core";

class QueryInput extends React.Component {
    constructor(props, context) {
        super(props, context);

        // Bind functions to object
        this.onSubmit = this.onSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.handleRequest = this.props.onRequest;

        // Initialize state
        this.state = {
            value: '',
        };
    }

    // Handle submit of sql command
    onSubmit(event) {
        // Prevent site from refreshing
        event.preventDefault();

        this.handleRequest(this.state.value);

        // Reset input
        this.setState({value: ''});
    }
    // Handles user input in query textfield
    handleChange(event){
        // Set state to input
        this.setState({value: event.target.value});
    }

    render() {
        return(
            <form onSubmit={this.onSubmit} noValidate>
                <FormControl onSubmit={this.onSubmit} variant="standard" fullWidth>
                        <TextField id='query-command'
                               fullWidth
                               placeholder='PostgreSQL Query Command'
                               aria-describedby='query-button'
                               value={this.state.value}
                               onChange={this.handleChange}
                                   variant="outlined"
                               InputProps={{
                                       endAdornment: (<InputAdornment>
                                           <Button color='primary' variant="contained" onClick={this.onSubmit} type='button'>Run</Button>
                                       </InputAdornment>),
                                   style: {
                                           fontSize: "large"
                                   },
                                   }}
                               autoFocus/>
                </FormControl>
            </form>
        );
    }
}

export default QueryInput;