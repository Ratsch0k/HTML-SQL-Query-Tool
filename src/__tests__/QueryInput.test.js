import React from 'react';
import ReactDom from 'react-dom';
import QueryInput from '../QueryInput';
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render} from "react-testing-library";

afterEach(cleanup);

it('QueryInput renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<QueryInput />, div);
    ReactDom.unmountComponentAtNode(div);
});

it('input changes when submitted', () => {
    const testInput = 'testInput';
    const mockUpParentFunction = (value) => {
        expect(value).toBe(testInput);
    };

    const component = renderer.create(<QueryInput onRequest={mockUpParentFunction}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // Click submit button and check if matches snapshot
    // Mock event with preventDefault
    let mockEvent = {
        preventDefault() {
            return null;
        },
        target: {
            value: testInput,
        },
    };

    tree.children[0].children[0].props.onChange(mockEvent);
    tree.props.onSubmit(mockEvent);
    // re-render
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});