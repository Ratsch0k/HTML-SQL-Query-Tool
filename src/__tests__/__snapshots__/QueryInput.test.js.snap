// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input changes when submitted 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="mb-3 mt-5 input-group input-group-lg"
    id="query-input"
  >
    <input
      aria-describedby="query-button"
      autoFocus={true}
      className="form-control"
      id="query-command"
      onChange={[Function]}
      placeholder="PostgreSQL Query Command"
      value=""
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-outline-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Run
      </button>
    </div>
  </div>
</form>
`;

exports[`input changes when submitted 2`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="mb-1 mt-1 input-group"
    id="query-input"
  >
    <input
      aria-describedby="query-button"
      autoFocus={true}
      className="form-control"
      id="query-command"
      onChange={[Function]}
      placeholder="PostgreSQL Query Command"
      value=""
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-outline-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Run
      </button>
    </div>
  </div>
</form>
`;
